airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -201,7 +199,13 @@ private Throwable getRootConfigError(final Exception e) {
   }
 
   private boolean isConfigError(final Throwable e) {
-    return e instanceof ConfigErrorException || e instanceof ConnectionErrorException;
+    return e instanceof ConfigErrorException || e instanceof ConnectionErrorException || isRecoveryConnectionException(e);
+  }
+
+  private boolean isRecoveryConnectionException(Throwable e) {
+    return e instanceof SQLException && e.getMessage()
+            .toLowerCase(Locale.ROOT)
+            .contains("terminating connection due to conflict with recovery");
```

Comment:
I think we'd need a better error message here. 
This is the information/root cause : https://docs.airbyte.com/integrations/sources/postgres/#sync-data-from-postgres-hot-standby-server

Trying to figure out if there is a better way to communicate this